### ISSUE TITLE
How to Create a Signed S3 File URL

### ISSUE THREAD

**Issue by amberaggarwal** [aws-amplify/aws-sdk-ios#41]
We just started using the new AWS SDK for iOS (2.0) and we are not able to figure out how we can get a signed url so that our customers can access the file they uploaded via URL.

We were able to create the signed url using the following code snippet in prior release of AWS SDK

```
S3ResponseHeaderOverrides *override = [[S3ResponseHeaderOverrides alloc] init];
override.contentType = asset.contentType;

S3GetPreSignedURLRequest *s3SignedUrl = [[S3GetPreSignedURLRequest alloc] init];
s3SignedUrl.key     = destinationImageName;
s3SignedUrl.bucket  = bucketName;
s3SignedUrl.expires = [NSDate dateWithTimeIntervalSinceNow:(NSTimeInterval) 604800];
s3SignedUrl.responseHeaderOverrides = override;

NSURL *url = [s3 getPreSignedURL:s3SignedUrl];
```

Please suggest what is the best way to do this in 2.0 release.

Thanks
AA

**Comment by winzig** [comment]
@yosuke-matsuda Earlier in the thread, in August, you stated "In general, it's not a good idea to generate Cloud Front private distribution URLs on mobile devices." Then in September you stated, "We've just released the AWS Mobile SDK for iOS 2.0.7, which contains AWSS3PreSignedURLBuilder."

Are you saying AWSS3PreSignedURLBuilder facilitates securely building pre-signed URLs for CloudFront, or should we still be creating those server-side? Thanks!

**Comment by yosuke-matsuda** [comment]
Amazon S3 PreSigned URL and Amazon Cloud Front are different. There is no way for us to generate a PreSigned URL for Cloud Front using the AWS credentials (it needs a Cloud Front certificate). You should continue generating the Cloud Front PreSigned URL on your server.
